test/prepare_disk.sh

```diff
@@ -0,0 +1,431 @@
+    # with new the new disk partition parameters defined above.
+    sed -i "s|${TARGET_STORAGE_DEVICE}p3.*type|${TARGET_STORAGE_DEVICE}p3 : start= ${USERDATA_START}, size= ${new_userdata_size}, type|" "${PARTITION_TABLE_FILE}"
+
+    execute_prepare_disk || return 0
+}
+
+test_corrupted_ext4_primary_superblock_ok()
+{
+    checksum_size="4"
+    block_size="1024"
+    primary_superblock_start="1"
+    write_start_offset="$((primary_superblock_start * block_size + checksum_size))"
+    dd if=/dev/zero of="${TARGET_STORAGE_DEVICE}p1" bs=1 count=10 seek="${write_start_offset}"
+
+    test_grow_boot_ok || return 1
+}
+
+test_corrupted_f2fs_primary_superblock_ok()
+{
+    # f2fs superblock are located in the beginning of the filesystem, destroy primary
+    f2fs_superblock_size="$((BYTES_PER_SECTOR * 10))"
+    dd if=/dev/urandom of="${TARGET_STORAGE_DEVICE}p2" bs=1 count="$((f2fs_superblock_size / 2))"
+
+    test_grow_rootfs_ok || return 1
+}
+
+test_corrupted_f2fs_superblocks_ok()
+{
+    # f2fs superblock are located in the beginning of the filesystem, destroy the primary and secondary
+    f2fs_superblock_size="$((BYTES_PER_SECTOR * 10))"
+    dd if=/dev/urandom of="${TARGET_STORAGE_DEVICE}p3" bs=1 count="$((f2fs_superblock_size + 1024))"
+
+    test_grow_rootfs_ok || return 1
+}
+
+test_partition_table_file_does_not_exist_nok()
+{
+    chroot_environment="TARGET_STORAGE_DEVICE=${TARGET_STORAGE_DEVICE}"
+    chroot "${toolbox_root_dir}" /bin/sh -c "${PREPARE_DISK_COMMAND}" || return 0
+}
+
+test_partition_table_file_does_not_exist_in_given_system_update_dir_nok()
+{
+    chroot_environment="TARGET_STORAGE_DEVICE=${TARGET_STORAGE_DEVICE}"
+    chroot_environment="${chroot_environment} SYSTEM_UPDATE_DIR=/etc"
+    chroot "${toolbox_root_dir}" /bin/sh -c "${chroot_environment} ${PREPARE_DISK_COMMAND}" || return 0
+}
+
+test_target_storage_device_argument_not_provided_nok()
+{
+    chroot "${toolbox_root_dir}" /bin/sh -c "${PREPARE_DISK_COMMAND}" || return 0
+}
+
+test_target_storage_device_is_not_a_block_device_nok()
+{
+    chroot_environment="TARGET_STORAGE_DEVICE=/dev/tty1"
+    chroot "${toolbox_root_dir}" /bin/sh -c "${chroot_environment} ${PREPARE_DISK_COMMAND}" || return 0
+}
+
+usage()
+{
+    echo "Usage:   ${0} [OPTIONS] <toolbox image file>"
+    echo "  -e   Stop consecutive tests on failure without cleanup"
+    echo "  -h   Print usage"
+    echo "NOTE: This script requires root permissions to run."
+}
+
+while getopts ":eh" options; do
+    case "${options}" in
+    e)
+        exit_on_failure=true
+        ;;
+    h)
+        usage
+        exit 0
+        ;;
+    :)
+        echo "Option -${OPTARG} requires an argument."
+        exit 1
+        ;;
+    ?)
+        echo "Invalid option: -${OPTARG}."
+        exit 1
+        ;;
+    esac
+done
+shift "$((OPTIND - 1))"
+
+if [ "${#}" -ne 1 ]; then
+    echo "Missing argument <file.img>."
```

Comment:
Hmm, only spotted this by accident here, but did you not rename this before?

Reply:
I changed it in the usage, not in this place

Reply:
Not sure this was changed here, EMP-406 did change it elsewhere I think